raft/server.cc

```diff
@@ -682,10 +682,10 @@ future<add_entry_reply> server_impl::execute_modify_config(server_id from,
         if (const auto* ex = dynamic_cast<const not_a_leader*>(&e)) {
             co_return add_entry_reply{transient_error{std::current_exception(), ex->leader}};
         }
-        if (const auto* ex = dynamic_cast<const dropped_entry*>(&e)) {
+        if (dynamic_cast<const dropped_entry*>(&e)) {
```

Comment:
we could have multiple catch blocks for dispatching based on the exact type of `e`. so we don't have to use `dynamic_cast<>` here. IMHO, RTTI should be used only if it is necessary. but that's not in the scope of this change. i will create a follow-up change to implement this.

Reply:
#13325